Tests/test_SeqIO.py

```diff
@@ -61,12 +62,16 @@ def get_mode(cls, fmt):
         if mode is not None:
             return mode
         for mode, stream in (("t", StringIO()), ("b", BytesIO())):
-            try:
-                SeqIO.read(stream, fmt)
-            except StreamModeError:
-                continue
-            except ValueError:  # SeqIO.read will complain that the stream is empty
-                pass
+            with warnings.catch_warnings():
+                warnings.simplefilter("error")
+                try:
+                    SeqIO.read(stream, fmt)
+                except StreamModeError:
+                    continue
+                except BiopythonDeprecationWarning:  # uniprot-xml
+                    continue
+                except ValueError:  # SeqIO.read will complain that the stream is empty
```

Comment:
Can you confirm the exception message matches here. Possibly one of the unittest assert raises methods will cover this.

Reply:
Done. Note that this is not a test, but a helper function to figure out the right mode (text or binary) to use.